akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -882,6 +882,46 @@ trait FlowOps[+Out, +Mat] {
   def recoverWith[T >: Out](pf: PartialFunction[Throwable, Graph[SourceShape[T], NotUsed]]): Repr[T] =
     via(new RecoverWith(-1, pf))
 
+  /**
+   * CompleteOn allows to complete the stream when an upstream error occurs.
+   *
+   * Since the underlying failure signal onError arrives out-of-band, it might jump over existing elements.
+   * This operator can recover the failure signal, but not the skipped elements, which will be dropped.
+   *
+   * '''Emits when''' element is available from the upstream
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes or fails with exception is an instance of the provided type
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def completeOn[T <: Throwable](implicit tag: ClassTag[T]): Repr[Out] = completeOn {
```

Comment:
should I change it to 
```scala
def completeOn()[T <: Throwable](implicit tag: ClassTag[T])
```
otherwise not compile on 2.12.x

Reply:
I guess that might be fine, `.completeOn[SomeEx]()` doesn't look too bad IMO

Reply:
Or rewrite it as 
```
    val pf: PartialFunction[Throwable, Boolean] = {
      case ex if tag.runtimeClass.isInstance(ex) => true
    }
    completeOn(pf)
```
?

Reply:
Yeah, https://github.com/scala/bug/issues/12688, seems the only way to go now is  :
```
def completeOn()[T <: Throwable](implicit tag: ClassTag[T])
```